### ISSUE TITLE
apply alignment to profiles

### ISSUE THREAD

**Issue by fmaumusINRA** [soedinglab/MMseqs2#388]
Hi,

I am trying to turn clusters of nucleotide sequences into profiles.
I did the following:
mmseqs cluster Athself AthselfDB /tmp -c 0.5
mmseqs createseqfiledb Athself AthselfDB AthselfDBSEQ --min-sequences 5
nohup mmseqs apply AthselfDBSEQ AthselfDBSEQ_msa -- mafft --adjustdirection - 1>& aln.out &

It ends up with AthselfDBSEQ_msa which apparently contains a cat of all output aligments.
Next, I would like to generate profiles for each cluster from this output, is that possible?

Thanks!

**Comment by milot-mirdita** [comment]
I noticed only after I wrote my answer below that you are trying to use nucleotides for profiles. We don't support that, as nulceotide searches below 80% sequence identity don't tend to be very useful, and normal sequence search is more than sufficient for that.

-----

If you had protein input you could do this:

If each database entry is a normal aligned FASTA MSA, then you should be able to give this database to `msa2profile`.
`msa2profile` expects a special database type (11 for MSA db) though:
```
awk 'BEGIN { printf("%c%c%c%c",11,0,0,0); exit; }' > AthselfDBSEQ_msa.dbtype 
mmseqs msa2profile AthselfDBSEQ_msa AthselfDBSEQ_profile ...
```
Usually you don't want global MSAs for profile computation though. Generally from experience, local alignments to a center sequence perform better.

**Comment by fmaumusINRA** [comment]
Thanks a lot for your reply, 
Your answer was very helpful
I am actually trying to see how mmseqs could be useful to identify repetitive elements in eukaryotic genomes.
So I am building clusters of highly similar sequences obtained by search against self to derive a representative profiles.

I have a related problem but could open a new issue for this if suitable:
When using "search", the max hit length I obtain is 10kb, even when forcing the max seq len parameter. Is there a way to obtain longer hits?

Kind regards
Florian

**Comment by milot-mirdita** [comment]
Did you create an index for the target database? If so, you have to give the max sequence `--max-seq-len` also to the `create index` invocation. That should be the correct parameter. Could you post the full calls to MMseqs2? Increasing this parameter will make the alignment potentially slower/raise their memory usage.

The currently existing code infrastructure would be not very far off to support it, however it's out of our current scope. If you have any bachelor/master students willing to take on this project we could help out with that project.

**Comment by fmaumusINRA** [comment]
Thanks a lot for offering your help, we will check how to make this a project. It's really kind of you.

I have tried indexing with --max-seq-len 30000 but the search then run into error:
mmseqs createindex Ath /tmp/ --search-type 3 --max-seq-len 30000
mmseqs search Ath Ath AthDB /tmp --search-type 3 --max-seq-len 30000
Writing results to: AthDB
terminate called after throwing an instance of 'std::logic_error' ] 0.00% 1 eta -
  what():  basic_string::_M_construct null not valid
/tmp/16276944363020795345/blastn.sh: line 66: 15898 Aborted                 "$MMSEQS" offsetalignment "$1" "${QUERY}" "$2" "${TARGET}" "$4/aln" "$3" ${OFFSETALIGNMENT_PAR}
Error: Offset step died

Ath, being the genome database (query and target)

**Comment by fmaumusINRA** [comment]
--max-seq-len works fine now, thanks.

Now I have run the command 
mmseqs msa2profile AthselfDBSEQ_msa AthselfDBSEQ_profile 
Is it possible to use the resulting profile file as a query or target for homology search with mmseqs or in hh-suite?

**Comment by milot-mirdita** [comment]
For MMseqs2 it's easy you can just pass this database as the query database:
```
mmseqs search AthselfDBSEQ_profile TARGETDB resdb tmp
mmseqs convertalis AthselfDBSEQ_profile TARGETDB resdb.m8
```

For hhblits you should be able to directly use `AthselfDBSEQ_msa` with e.g. `hhblits_omp` which can take a database as input instead of single files.

**Comment by fmaumusINRA** [comment]
I have tried the search against nucleotide db with the following error:
 mmseqs search AthselfDBSEQ_profile Ath resdb tmp
Invalid input database and --search-type combination
queryDbType: Profile
targetDbType: Nucleotide
targetSrcDbType: Nucleotide
searchMode: 0

or with --search-type 3:
mmseqs search AthselfDBSEQ_profile Ath resdb tmp --search-type 3
Invalid input database and --search-type combination
queryDbType: Profile
targetDbType: Nucleotide
targetSrcDbType: Nucleotide
searchMode: 3

Is there any specific search-type when using profile against dna?

**Comment by milot-mirdita** [comment]
Sorry, I misunderstood you. Any combination of profiles and nucleotides will not currently.

Adding nucleotide profile support to HHblits would be a far more difficult task than doing that in MMseqs2. I would not recommend going that route.
